site/content/integrate/apps/_index.md

```diff
@@ -33,53 +33,53 @@ When you develop using the Apps Framework, your apps can:
 
 Apps are now generally available and we always appreciate your feedback! Share constructive feedback in the [Mattermost Apps channel](https://community.mattermost.com/core/channels/mattermost-apps) on our Mattermost community instance.
 
-Read the [JavaScript start guide]({{< ref "quickstart/quick-start-js" >}}) or [Go quick start guide]({{< ref "quickstart/quick-start-go" >}}) to learn how to write your first App.
+Read the quick start guides for [Hello World in TypeScript]({{< ref "quickstart/quick-start-ts" >}}) or [Hello World in Go]({{< ref "quickstart/quick-start-go" >}}) to learn how to write your first App.
 
 ## FAQ
 
-### When would you build an app vs. a custom slash command vs. a webhook vs. a plugin?
+### When would you build an App vs. a custom slash command vs. a webhook vs. a plugin?
 
 That depends on your use case, as they each have benefits.
 
-The built-in [incoming webhook]({{< ref "/integrate/webhooks/incoming" >}}) is great for simple use cases. It requires the incoming payload to contain a valid {{< newtabref href="https://pkg.go.dev/github.com/nhannv/mattermost-server/model#Post" title="Post" >}} JSON object, but some external systems don't allow customization of the fields included in the webhook payload. Compared to built-in webhooks, an App or plugin can be customized to receive an HTTP webhook posting from another system, and can process the incoming data then augment it or make an actionable message in a channel.
+The built-in [incoming webhook]({{< ref "/integrate/webhooks/incoming" >}}) is great for simple use cases. It requires the incoming payload to contain a valid {{< newtabref href="https://pkg.go.dev/github.com/mattermost/mattermost-server/v6/model#Post" title="Post" >}} JSON object, but some external systems don't allow customization of the fields included in the webhook payload. Compared to built-in webhooks, an App or plugin can be customized to receive an HTTP webhook posting from another system, and can process the incoming data then augment it or make an actionable message in a channel.
```

Comment:
I (selfishly) like the way I worded this [here](https://community.mattermost.com/core/pl/4jtbfepncigyfxxfnfnycbfn1c)

> Each type of existing integration type is sort of a la carte, whereas the App framework is an all-in-one package that supports its own version of some of those out of the box. Meaning when you install an App, you don't need to go create a separate bot account, OAuth app, webhooks etc. Some pieces are reused, such as the bot account mechanism, but some pieces were remade and are not reused, such as webhooks.
>
> In the case of webhooks, the existing mechanism is only able to create posts, and only accepts a webhook payload that matches a Mattermost post's data structure. It does not support other logic for handling arbitrary data structures from external systems.

Just some additional context we can provide here. Also, should we have a table of contents on this page so this section is easier to find? Or maybe have its own more discoverable page, that is agnostic to a specific integration type?

Reply:
I remember this wording -- thanks for bringing this back up. I've incorporated it here. Like I mentioned in my other comment, I think a revamp of this page would be a future PR. There's a lot to sort out here and we should be more clear on the goals of the page.